abci/types/application.go

```diff
@@ -19,10 +19,10 @@ type Application interface {
 	// Consensus Connection
 	InitChain(RequestInitChain) ResponseInitChain // Initialize blockchain w validators/other info from TendermintCore
 	PrepareProposal(RequestPrepareProposal) ResponsePrepareProposal
-	BeginBlock(RequestBeginBlock) ResponseBeginBlock // Signals the beginning of a block
-	DeliverTx(RequestDeliverTx) ResponseDeliverTx    // Deliver a tx for full processing
-	EndBlock(RequestEndBlock) ResponseEndBlock       // Signals the end of a block, returns changes to the validator set
-	Commit() ResponseCommit                          // Commit the state and return the application Merkle root hash
+	BeginBlock(RequestBeginBlock) ResponseBeginBlock                            // Signals the beginning of a block
```

Comment:
Since this is an interface describing the API, I would place actual godocs above the method names and we can omit godocs on an implementations in Tendermint.